src/olympia/devhub/tests/test_forms.py

```diff
@@ -929,25 +932,71 @@ def test_site_permission_generator_origin_invalid(origin):
         'https://example.com:8888',
     ],
 )
+@pytest.mark.django_db
 def test_site_permission_generator_origin_valid(origin):
+    request = req_factory_factory('/', user=user_factory())
     form = forms.SitePermissionGeneratorForm(
-        {'site_permissions': _DEFAULT_SITE_PERMISSIONS, 'origin': origin}
+        {'site_permissions': _DEFAULT_SITE_PERMISSIONS, 'origin': origin},
+        request=request,
     )
     assert form.is_valid()
     assert form.cleaned_data['origin'] == origin
 
 
 @pytest.mark.django_db
 def test_site_permission_generator_origin_denied():
+    request = req_factory_factory('/', user=user_factory())
     DeniedInstallOrigin.objects.create(hostname_pattern='*.tld')
     form = forms.SitePermissionGeneratorForm(
-        {'site_permissions': _DEFAULT_SITE_PERMISSIONS, 'origin': 'https://foo.com'}
+        {'site_permissions': _DEFAULT_SITE_PERMISSIONS, 'origin': 'https://foo.com'},
+        request=request,
     )
     assert form.is_valid()
     form = forms.SitePermissionGeneratorForm(
-        {'site_permissions': _DEFAULT_SITE_PERMISSIONS, 'origin': 'https://foo.tld'}
+        {'site_permissions': _DEFAULT_SITE_PERMISSIONS, 'origin': 'https://foo.tld'},
+        request=request,
     )
     assert not form.is_valid()
     assert form.errors['origin'] == [
         'The install origin https://foo.tld is not permitted.'
     ]
+
+
+@pytest.mark.django_db
+def test_site_permission_generator_duplicates():
+    request = req_factory_factory('/', user=user_factory())
+    addon = addon_factory(type=amo.ADDON_SITE_PERMISSION, users=[request.user])
+    version = addon.versions.get()
+    FileSitePermission.objects.create(file=version.file, permissions=['something'])
+    version.installorigin_set.create(origin='https://example.com')
+
+    # User already has a site permission, but for a different origin/permissions.
+    form = forms.SitePermissionGeneratorForm(
+        {'site_permissions': _DEFAULT_SITE_PERMISSIONS, 'origin': 'https://foo.com'},
+        request=request,
+    )
+    assert form.is_valid()
+
+    # User already has a site permission, but for different permissions
```

Comment:
There's a test for different origin/different permissions, and a test for same origin/different permissions, but there isn't a test for different origin/same permissions. Should there be?

Reply:
Yeah, was a bit lazy, added in fe9f7bfef82ad38741f9fca7bf896cba00f6bc6c